cql3/statements/describe_statement.cc

```diff
@@ -0,0 +1,535 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "cql3/statements/prepared_statement.hh"
+#include "db/consistency_level_type.hh"
+#include "db/timeout_clock.hh"
+#include "dht/i_partitioner.hh"
+#include "exceptions/exceptions.hh"
+#include "keys.hh"
+#include "partition_slice_builder.hh"
+#include "query-request.hh"
+#include "types.hh"
+#include "cql3/query_processor.hh"
+#include "cql3/cql_statement.hh"
+#include "cql3/statements/raw/describe_statement.hh"
+#include "cql3/statements/describe_statement.hh"
+#include "seastar/core/shared_ptr.hh"
+#include "transport/messages/result_message.hh"
+#include "transport/messages/result_message_base.hh"
+#include "service/query_state.hh"
+#include "data_dictionary/keyspace_metadata.hh"
+#include "data_dictionary/data_dictionary.hh"
+#include <boost/range/adaptor/transformed.hpp>
+#include <boost/range/algorithm/sort.hpp>
+#include <memory>
+#include <optional>
+#include "db/system_keyspace.hh"
+#include "service/storage_proxy.hh"
+#include "cql3/ut_name.hh"
+
+bool is_internal_keyspace(std::string_view name);
+
+namespace cql3 {
+
+namespace statements {
+
+static const sstring KS = "system";
+static const sstring CF = "describe";
```

Comment:
I'm using KS to find schema of those tables and CF is used when I'm manually creating result metadata (only in `DESC CLUSTER`)

I have to create it manually there because the cql_statement::get_result_metadata has signature: `seastar::shared_ptr<const metadata> get_result_metadata() const` and cluster description depends on client_state (whether the keyspace is set to user keyspace, then the token range to endpoint column is added). In other cases, there is predefined selection, from which the result metadata is taken.

Reply:
I still don't understand what these definitions do at this point. I think you need to move them to where they are used and perhaps be clearer - or add a comment.

Reply:
Since it appears only a few times in the whole file, I've removed those variables and directly used those strings. I didn't know how to nicely describe them and they didn't deserve that much attention